index.html

```diff
@@ -1009,20 +1016,36 @@
             <var>b</var> as a <a>USVString</a>. The <a>user agent</a> MUST use a serialization
             method that does not branch based on the value of <var>b</var>.
             </li>
-            <li>Add an entry to <var>keys</var> whose key name is the name of <var>i</var> and
-            whose value is <var>s</var>.
+            <li>Add an entry to the <a data-link-for="PushSubscriptionJSON">keys</a> property of
+            <var>json</var> whose key is the name of <var>i</var> and whose value is <var>s</var>.
             </li>
           </ol>
         </li>
-        <li>Add an entry to <var>map</var> whose key name is <code>keys</code> and whose value is
-        <var>keys</var>.
-        </li>
-        <li>Return <var>map</var>.
-        </li>
+        <li>Return <var>json</var>.
       </ol>
+
+      <p>
+        A <dfn>PushSubscriptionJSON</dfn> dictionary represents the <a>JSON type</a> of a
+        <a>PushSubscription</a> that can be converted into a JSON string.
+      </p>
+      <p>
+        The <dfn data-dfn-for="PushSubscriptionJSON">endpoint</dfn> contains the serialized value of
+        the <a data-link-for="PushSubscription">endpoint</a> attribute.
+      </p>
+      <p>
+        The <dfn data-dfn-for="PushSubscriptionJSON">expirationTime</dfn> contains the serialized
+        value of the <a data-link-for="PushSubscription">expirationTime</a> attribute.
+      </p>
+      <p>
+        The <dfn data-dfn-for="PushSubscriptionJSON">keys</dfn> contains a mapping from each
```

Comment:
I'd replace the above by:

```
The <dfn data-dfn-for="PushSubscriptionJSON">keys</dfn> record contains an entry for each
```
(see https://infra.spec.whatwg.org/#map-entry if you want to reference the term properly.)